pkg/terraform/gcp/terraform_gcp.go

```diff
@@ -130,79 +148,108 @@ func createCustomTokens(tokenName string) hclwrite.Tokens {
+		gcpInstanceBody.SetAttributeValue("machine_type", cty.StringVal(instanceType))
+		metadataMap := make(map[string]cty.Value)
+		metadataMap["ssh-keys"] = cty.StringVal(fmt.Sprintf("ubuntu:%s", strings.TrimSuffix(sshPublicKey, "\n")))
+		gcpInstanceBody.SetAttributeValue("metadata", cty.ObjectVal(metadataMap))
+		networkInterface := gcpInstanceBody.AppendNewBlock("network_interface", []string{})
+		networkInterfaceBody := networkInterface.Body()
+		networkRoot := "google_compute_network"
+		if networkExists {
+			networkRoot = "data.google_compute_network"
+		}
+		networkInterfaceBody.SetAttributeTraversal("network", hcl.Traversal{
+			hcl.TraverseRoot{
+				Name: networkRoot,
+			},
+			hcl.TraverseAttr{
+				Name: networkName,
 			},
 			hcl.TraverseAttr{
-				Name: "address",
+				Name: "id",
 			},
 		})
+		accessConfig := networkInterfaceBody.AppendNewBlock("access_config", []string{})
+		// don't add google_compute_address if user is not using public IP
+		if staticIPName[zone] != "" {
+			accessConfigBody := accessConfig.Body()
+			accessConfigBody.SetAttributeTraversal("nat_ip", hcl.Traversal{
+				hcl.TraverseRoot{
+					Name: "google_compute_address",
+				},
+				hcl.TraverseAttr{
+					Name: fmt.Sprintf("%s[count.index]", terraformStaticIPName(nodeName[zone])),
+				},
+				hcl.TraverseAttr{
+					Name: "address",
+				},
+			})
+		}
+		bootDisk := gcpInstanceBody.AppendNewBlock("boot_disk", []string{})
+		bootDiskBody := bootDisk.Body()
+		initParams := bootDiskBody.AppendNewBlock("initialize_params", []string{})
+		initParamsBody := initParams.Body()
+		initParamsBody.SetAttributeValue("image", cty.StringVal(ami))
+		initParamsBody.SetAttributeValue("size", cty.NumberIntVal(1000))
+		gcpInstanceBody.SetAttributeValue("allow_stopping_for_update", cty.BoolVal(true))
 	}
-	bootDisk := gcpInstanceBody.AppendNewBlock("boot_disk", []string{})
-	bootDiskBody := bootDisk.Body()
-	initParams := bootDiskBody.AppendNewBlock("initialize_params", []string{})
-	initParamsBody := initParams.Body()
-	initParamsBody.SetAttributeValue("image", cty.StringVal(ami))
-	initParamsBody.SetAttributeValue("size", cty.NumberIntVal(1000))
-
-	gcpInstanceBody.SetAttributeValue("allow_stopping_for_update", cty.BoolVal(true))
 }
 
 // SetOutput adds output section in terraform state file so that we can call terraform output command and print instance_ip and instance_id to user
-func SetOutput(rootBody *hclwrite.Body) {
-	outputEip := rootBody.AppendNewBlock("output", []string{"instance_ips"})
-	outputEipBody := outputEip.Body()
-	outputEipBody.SetAttributeTraversal("value", hcl.Traversal{
-		hcl.TraverseRoot{
-			Name: "google_compute_instance",
-		},
-		hcl.TraverseAttr{
-			Name: "gcp-node[*]",
-		},
-		hcl.TraverseAttr{
-			Name: "network_interface.0.access_config.0.nat_ip",
-		},
-	})
+func SetOutput(rootBody *hclwrite.Body, zones []string) {
+	for _, zone := range zones {
+		outputEip := rootBody.AppendNewBlock("output", []string{"instance_ips_" + zone})
```

Comment:
why do you want to do this per zone?

Reply:
because they have unique names for tf resources